### PR TITLE
Fixed 76 QueryParser Failing Tests

### PR DESCRIPTION
**Pull request by NightOwl888**
This fixes 76 of the 77 failing tests in QueryParser. The remaining one is a matter of completing the test implementation (or removing it entirely depending on exactly how to handle the TimeZone).

### CODE REVIEW

**Comment by synhershko** [comment]
Let's take the discussion on TimeZone to the dev@ mailing list - can you please start a thread there explaining the issue / dilemma and I will follow up?

File: src/Lucene.Net.QueryParser/Surround/Parser/QueryParser.cs [hunk 1]
```diff
@@ -883,7 +883,7 @@ private void Jj_rescan_token()
                         p = p.next;
                     } while (p != null);
                 }
-                catch (LookaheadSuccess ls) { }
+                catch (LookaheadSuccess /*ls*/) { }
             }
             jj_rescan = false;
         }
```
Comment (synhershko) [inline src/Lucene.Net.QueryParser/Surround/Parser/QueryParser.cs]: this file is auto-generated in Java so I'm not sure whether or not it's a good idea to change it here...

**Comment by synhershko** [comment]
Great job, merging this now

File: src/Lucene.Net.QueryParser/Surround/Parser/QueryParser.cs (hunk 1 above)
Comment (NightOwl888) [inline src/Lucene.Net.QueryParser/Surround/Parser/QueryParser.cs]: Well, the alternative would be to use `#pragma` to disable the warning messages for the unused exception variable. Would you prefer it that way?

Comment (synhershko) [inline src/Lucene.Net.QueryParser/Surround/Parser/QueryParser.cs]: Yes :)

**Pull request merged**